backend/runner/runner.go

```diff
@@ -173,12 +177,13 @@ func (s *Service) Deploy(ctx context.Context, req *connect.Request[ftlv1.DeployR
 			return nil, errors.Wrap(err, "failed to create deployment directory")
 		}
 	}
-	err = download.Artefacts(ctx, s.controllerClient, id, deploymentDir)
+	err = download.Artefacts(ctx, s.controllerClient, key, deploymentDir)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to download artefacts")
 	}
 	deployment, cmdCtx, err := plugin.Spawn(
 		unstoppable.Context(ctx),
+		logger,
```

Comment:
You don't need to do this, use `logger.Withcontext(ctx, logger)` to add the new logger to a new context child.